src/taxonomy/TaxonomyListPage.jsx

```diff
@@ -1,29 +1,65 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import {
   CardView,
   Container,
   DataTable,
   Spinner,
+  Toast,
 } from '@edx/paragon';
 import { useIntl } from '@edx/frontend-platform/i18n';
+import { useSearchParams } from 'react-router-dom';
 import { Helmet } from 'react-helmet';
 
 import SubHeader from '../generic/sub-header/SubHeader';
 import getPageHeadTitle from '../generic/utils';
 import messages from './messages';
 import TaxonomyCard from './taxonomy-card';
-import { useTaxonomyListDataResponse, useIsTaxonomyListDataLoaded } from './data/apiHooks';
+import { useTaxonomyListDataResponse, useIsTaxonomyListDataLoaded, useDeleteTaxonomy } from './data/apiHooks';
 
 const TaxonomyListPage = () => {
   const intl = useIntl();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [showToast, setShowToast] = useState(false);
+  const [toastText, setToastText] = useState('');
+  const deleteMutation = useDeleteTaxonomy();
+
+  const onDeleteTaxonomy = (id, name) => {
+    setToastText(intl.formatMessage(messages.taxonomyDeleteToast, { name }));
+    deleteMutation.mutate({ pk: id });
+  };
+
   const useTaxonomyListData = () => {
     const taxonomyListData = useTaxonomyListDataResponse();
     const isLoaded = useIsTaxonomyListDataLoaded();
+    if (isLoaded) {
+      for (let i = 0; i < taxonomyListData.results.length; i++) {
+        taxonomyListData.results[i].onDeleteTaxonomy = onDeleteTaxonomy;
+      }
+    }
```

Comment:
I'm not sure what this is accomplishing. The same function is being used for all results, so the function can just be called directly?

Reply:
Thanks, I found another way to transfer more data to the Card. This change is here: https://github.com/openedx/frontend-app-course-authoring/pull/684/commits/0c6f63b2f7c921a737e51796e05020e579c5f990